### PR TITLE
wrong comparison

### LINKED ISSUE #15: Not defined variable in fnc_addChestpack.sqf
**Arma 3 Version:** `2.16.151618` (stable)
**CBA Version:** `3.17.1.240424` (stable)
**ACE3 Version:** `3.14.0` (stable)

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- calling bocr_main_fnc_addChestpack function returns error for not defining variable _backpackload in 52 line of fnc_addChestpack.sqf func.

**Steps to reproduce:**
- calling function **bocr_main_fnc_addChestpack** without sending last argument (_backpackLoad);

**Where did the issue occur?**
- Self-Hosted Multiplayer

**Additional information:**
- no information

### PR DESCRIPTION
**Pull request by Canibus**
**When merged this pull request will:**
- fix #15

### CODE REVIEW

File: addons/main/functions/fnc_addChestpack.sqf [hunk 1]
```diff
@@ -17,9 +17,9 @@
  *
  * Public: No
  */
-params ["_unit", "_chestpackClass", ["_backpackLoadout", []], ["_backpackVariables", []], ["_backpackLoad", nil]];
+params ["_unit", "_chestpackClass", ["_backpackLoadout", []], ["_backpackVariables", []], ["_backpackLoad", -1]];
 
-if (_backpackLoad isEqualTo nil) then {
+if (_backpackLoad isEqualTo -1) then {
```
Comment (rautamiekka) [inline addons/main/functions/fnc_addChestpack.sqf:22]: A `< 0` would make it much more foolproof while indeed using the `-1` as the `params` undefined substitution.

Also, at least when I tried, `iET` was consistently slower for a number than `==`.

```suggestion
if (_backpackLoad < 0) then {
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]

File: addons/main/functions/fnc_addChestpack.sqf (hunk 1 above)
Comment (mjc4wilton) [reply addons/main/functions/fnc_addChestpack.sqf:22]: > Also, at least when I tried, `iET` was consistently slower for a number than `==`.

That should not be the case. `isEqualTo` is a strict, binary comparison which should be extremely efficient and also makes it significantly stricter than `==`.

File: addons/main/functions/fnc_addChestpack.sqf [hunk 2]
```diff
@@ -17,9 +17,9 @@
  *
  * Public: No
  */
-params ["_unit", "_chestpackClass", ["_backpackLoadout", []], ["_backpackVariables", []], ["_backpackLoad", nil]];
+params ["_unit", "_chestpackClass", ["_backpackLoadout", []], ["_backpackVariables", []], ["_backpackLoad", -1]];
```
Comment (mjc4wilton) [inline addons/main/functions/fnc_addChestpack.sqf:20]:
```suggestion
params ["_unit", "_chestpackClass", ["_backpackLoadout", []], ["_backpackVariables", []], ["_backpackLoad", -2]];
```

File: addons/main/functions/fnc_addChestpack.sqf (hunk 1 above)
Comment (mjc4wilton) [inline addons/main/functions/fnc_addChestpack.sqf:22]:
```suggestion
if (_backpackLoad isEqualTo -2) then {
```

**Review by mjc4wilton — CHANGES_REQUESTED** [review]
Lets switch this to -2 as the undefined case. Its really weird, but I want -1 to be reserved for an infinite load case.

File: addons/main/functions/fnc_addChestpack.sqf (hunk 1 above)
Comment (Canibus) [reply addons/main/functions/fnc_addChestpack.sqf]: > > Also, at least when I tried, `iET` was consistently slower for a number than `==`.
> 
> That should not be the case. `isEqualTo` is a strict, binary comparison which should be extremely efficient and also makes it significantly stricter than `==`.

totally agree

Comment (rautamiekka) [reply addons/main/functions/fnc_addChestpack.sqf]: Precisely my thinking, that's why I've been favoring `iET`/`iNET`, but I figured I'd test it. My machine has a consistently inconsistent performance (partially due to an incorrectly debloated Win10), especially with 'Arma 3', but somehow that's the result trying it about 20 times with both overall.

**Pull request merged**